common/djangoapps/student/views/dashboard.py

```diff
@@ -759,6 +773,18 @@ def student_dashboard(request):
             'ecommerce_payment_page': ecommerce_service.payment_page_url(),
         })
 
+    # Gather urls for resume buttons in course cards.
+    try:
+        resume_button_urls = _get_urls_for_resume_buttons(user, course_enrollments + course_entitlements)
```

Comment:
Could you please explain why we need to include `course_entitlements`? If the user has course_entitlements, and no enrollment, it should mean the learner hasn't accessed course yet, correct?

Reply:
The dashboard template creates course cards for enrollments and entitlements. It loops through the list of enrollments and entitlements and uses loop indices to assign values to specific cards. In order to prevent the template from indexing out of range, I provide resume button values for all course cards.

Reply:
@sandroroux Got it. Thanks for the explanation